packages/react/polymorphic/src/forwardRefWithAs.ts

```diff
@@ -49,6 +49,8 @@ interface ForwardRefExoticComponentWithAs<
   <As extends React.ElementType<any>>(
     props: MergeWithDOMProps<As, OwnProps & { as: As }>
   ): JSX.Element;
+
+  displayName: string;
```

Comment:
I've made this required on this polymorphic thing... I'm not sure this is a good idea tbh. It might be better to just do `CollapsibleContent.displayName!`. Thoughts?

Reply:
Wouldn't you get an error when you initially define the component because it's missing that property? This has never worked for me in the past, but maybe it's a tsconfig thing? 🧐

Reply:
> Wouldn't you get an error when you initially define the component

Yes, because I've made it required so I would expect that you would get an error telling you to add it. That's why I'm saying I don't think this is really a good idea here because some people might want to use this polymorphic stuff but don't care about `displayName`s.

I'm thinking the `CollapsibleContent.displayName!` approach would be better where I call `removePartDataAttrObj` but that also feels weird assuming we've correctly added the display name haha. 

Open to other ideas here of course!

Reply:
I think I'm with you. We shouldn't need this frequently so the occasional `!` is totes fine.

Reply:
The only thing I can think of is that some apps strip out `displayName` in their production builds to save a few bytes. No idea if those sort of tools should have an impact on `node_modules` or if it's something to consider, but it would break if that were in fact the case.

Reply:
LOL, good point! doh. Hmmm. I'm not sure either but I'll import the consts instead to be sure. Good spot :+1: